### PR TITLE
✨ Enhancement: adding new entries for GH actions & Pub as ecosystems, typo fixes

### PR DESCRIPTION
**Pull request by aidenwang9867**
#### What kind of change does this PR introduce?
1. Adding entry for GH actions as an ecosystem
2. Ecosystem naming convention mapping type fixes

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
The GitHub ecosystem `actions` (for the GitHub Action) is not an ecosystem defined by OSV. Missing this special entry in the convention map will lead to the entire `GetDependencyDiffCheckResults` API to fail.

#### What is the new behavior (if this is a feature change)?**
Adding the "action": "action" map entry for it not to fail.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?
No.

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]
LGTM.

File: dependencydiff/mapping.go [hunk 1]
```diff
@@ -78,12 +81,35 @@ var (
 		"composer": ecosystemPackagist,
 		"rubygems": ecosystemRubyGems,
 		"nuget":    ecosystemNuGet,
+		"actions":  ecosystemActions, /* created this entry for the "actions" not to fail in the mapping process. */
 	}
 )
 
+func mapDependencyEcosystemNaming(deps []dependency) error {
+	for i := range deps {
+		if deps[i].Ecosystem == nil {
```
Comment (laurentsimon) [inline dependencydiff/mapping.go:90]: why does this not return an error too?

Comment (aidenwang9867) [reply dependencydiff/mapping.go:90]: hmm, since we allow the package ecosystem field to be null-able?

Comment (laurentsimon) [reply dependencydiff/mapping.go:90]: ah, GH does not always populate it?
Let's add comments / notes to the code to explain.

Comment (aidenwang9867) [reply dependencydiff/mapping.go:90]: > ah, GH does not always populate it? Let's add comments / notes to the code to explain.

for the GitHub API as the data source, the `ecosystem` is a required field to return - but in our previous discussion we'd like to make this more generale and not limited by an specific API, so only we only made the `name` field a `string` (required).

i'll definitely add a comment here to explain

**Pull request merged**